libutils/buffer.c

```diff
@@ -333,7 +333,8 @@ void BufferAppendF(Buffer *buffer, const char *format, ...)
     va_copy(aq, ap);
 
     int printed = vsnprintf(buffer->buffer + buffer->used, buffer->capacity - buffer->used, format, aq);
-    if (printed >= (buffer->capacity - buffer->used))
+    assert(printed >= 0);
+    if ((unsigned int) printed >= (buffer->capacity - buffer->used))
```

Comment:
I'd prefer changing `buffer->capacity/used` to `size_t` and then work with a `size_t printed` variable here.

Reply:
Just be absolutely sure that `buffer->capacity/used` is never below 0, then(!). For example in while/for loops.

Reply:
@olehermanse  `buffer->capacity/used` is already `unsigned`. So that should not be a problem.